msgpack/src/main/scala/fs2/data/msgpack/low/internal/ItemParser.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data
+package msgpack
+package low
+package internal
+
+import fs2.data.msgpack.low.internal.Helpers._
+import fs2.data.msgpack.low.internal.FormatParsers._
+import scodec.bits._
+
+private[low] object ItemParser {
+
+  /** Parses a single item from the stream.
+    */
+  private def parseItem[F[_]](ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    requireOneByte(ctx) flatMap { res => {
+      val byte = res.result
+      val ctx = res.toContext
+
+      byte match {
```

Comment:
A nice way to have for sure something efficient, is to annotate it with `@switch`. This ensures that the emitted code is a switch. If it cannot be compiled as such it will emit a warning. For this to work, you will need to use `Int` instead of `Byte`. You can get it efficiently by doing:
```suggestion
      ((byte & 0xff): @switch) match {
```

And by removing all the `toByte` in the `Headers` object.

Reply:
Done in 3272920. I have actually tried to do this before but I missed the fact that `@switch` requires expression to be of type `Int` :smile:.